Automattic/jetpack
# My Jetpack: Fix Stats card status when not connected

## Proposed changes:

As the purchase page live in Odyssey Stats, which means `/add-stats` is empty. In order to walk around that, we redirect the `Start for free` button to the connection page when the site is not connected. We already dealt with that when a site is connected.

This is a temporary fix and we'll look through the integration between My Jetpack and Stats again and see what's the best we could do. 




### Other information:


## Jetpack product discussion
p1HpG7-pah-p2

## Does this pull request change what data or activity we track or use?
no

## Testing instructions:
* Spin up a fresh JN site
* Do not connect it
* Open `/wp-admin/admin.php?page=my-jetpack`
* Click 'Start for free' button from the Stats card
* Ensure you are directed to the connection page

<img width="371" alt="image" src="https://github.com/Automattic/jetpack/assets/1425433/9d65cd05-8911-41fe-86be-42d0e83874e7">
